src/main/java/com/microsoftopentechnologies/windowsazurestorage/helper/Utils.java

```diff
@@ -138,7 +143,11 @@ public static String getMD5(String plainText) {
             byte[] array = md.digest(plainText.getBytes(Charset.forName("UTF-8")));
             StringBuilder sb = new StringBuilder();
             for (int i = 0; i < array.length; ++i) {
-                sb.append(Integer.toHexString((array[i] & 0xFF) | 0x100).substring(1, 3));
+                String hex = Integer.toHexString(((int) array[i]) & BYTE_MASK);
+                if (hex.length() < 2) {
+                    sb.append('0');
+                }
+                sb.append(hex);
```

Comment:
Magic number to constants. I updated the logic a bit so as to reduce the magic numbers used.